### ISSUE TITLE
Cannot load class 'co.elastic.apm.attach.ElasticApmAttacher'

### ISSUE THREAD

**Issue by prco5** [michaelhyatt/elastic-apm-mule4-agent#13]
Hi,when starting mule app, i get the following error.
![image](https://user-images.githubusercontent.com/68184994/107526006-b2547a00-6bb7-11eb-985b-54b462ecb2ad.png)

I installed .jar correctly.Added <import> like this
![image](https://user-images.githubusercontent.com/68184994/107526193-e3cd4580-6bb7-11eb-939a-21548d50195f.png)
And then when i start app i get the error above.
![image](https://user-images.githubusercontent.com/68184994/107526391-18d99800-6bb8-11eb-9078-a08e026b7c7f.png)
Ty.

**Comment by michaelhyatt** [comment]
Can you try and get any of the `example` projects running in your environment? Try the `tracing-test` first to see that everything works correctly with your setup. There is also a domain example there.

**Comment by prco5** [comment]
Hi thank you for you answer.
So i installed jar like this.
![1111](https://user-images.githubusercontent.com/68184994/107741973-7e3f9d00-6d0e-11eb-8eb3-fc2db7a4d064.PNG)
Build success.
Imported project in studio.
![projectStrcutire](https://user-images.githubusercontent.com/68184994/107740344-323f2900-6d0b-11eb-9c07-8f13b19a4326.PNG)
![pom](https://user-images.githubusercontent.com/68184994/107740356-379c7380-6d0b-11eb-9db5-070dfb871f07.PNG)
I didnt do anything to import file, i left it just like that how i imported it.(tracer.xml)
And when i run the application i get the following.
![error](https://user-images.githubusercontent.com/68184994/107740449-70d4e380-6d0b-11eb-9e34-9c5f37cc02e8.PNG)
![error2](https://user-images.githubusercontent.com/68184994/107740453-73cfd400-6d0b-11eb-8b5c-534d4280187e.PNG)

Just to be clear im running this without the elastic.apm.config_file to mule agent? Is that maybe the source of the problem?
It's just a fresh maven install of jar, imported your tracing-test and just runing it.Am i missing something?
Ty.

**Comment by michaelhyatt** [comment]
Not having the config shouldn't cause this. Try to add this to the pom file:
```xml
<dependency>
    <groupId>co.elastic.apm</groupId>
    <artifactId>apm-agent-attach</artifactId>
    <version>1.17.0</version>
</dependency>
```

**Comment by prco5** [comment]
Yes with this added to pom.xml i'm able to start the application.Thank you very much.

![Capture](https://user-images.githubusercontent.com/68184994/107918539-29985e00-6f6a-11eb-9aa4-198a39916b0f.PNG)

But got issuse on server in following post.
Thank you for help.

**Comment by prco5** [comment]
Hi, also one thing, so i tried to deploy it to mule server running apm server,and there it can find server but  i get the following message in log file when i trigger the endpoint.

![11](https://user-images.githubusercontent.com/68184994/107926048-82b9bf00-6f75-11eb-8299-d05d30e51c70.PNG)

But funny thing in Kibana i can see some info in the following.
![image](https://user-images.githubusercontent.com/68184994/107927759-ba296b00-6f77-11eb-86cf-55e8cd500b8d.png)
But there is just some JVM server related stuff, not the actual mule app info.
But the log file is spammed with the above error.
Thank you.

**Comment by prco5** [comment]
Hi, got it running by adding
```
 <dependency>
    <groupId>co.elastic.apm</groupId>
    <artifactId>apm-agent-api</artifactId>
    <version>1.21.0</version>
</dependency>
```
Any way to see the actual payload or messages inside, or more data?
Thanks

**Comment by michaelhyatt** [comment]
Good to hear you managed to get it working. It doesn't capture message payloads or any additional labels, but it is possible to implement it in the code, similar to how Mule 3 agent captures message headers and properties.

TransactionUtils.java and SpanUtils.java have placeholders where such logic can reside.

**Comment by prco5** [comment]
Thank you for your help.
